packages/application-generic/src/encryption/encrypt-provider.ts

```diff
@@ -41,19 +44,34 @@ export function decryptCredentials(
 
   for (const key in credentials) {
     decryptedCredentials[key] =
-      typeof credentials[key] === 'string' &&
-      isEncryptedCredential(credentials[key])
-        ? decryptProviderSecret(credentials[key])
+      typeof credentials[key] === 'string' && novuEncrypted(credentials[key])
+        ? decryptSecret(credentials[key])
         : credentials[key];
   }
 
   return decryptedCredentials;
 }
 
-function isEncryptedCredential(text: string): boolean {
+export function encryptApiKey(apiKey: string): EncryptedSecret {
+  if (novuEncrypted(apiKey)) {
+    return apiKey;
+  }
+
+  return encryptSecret(apiKey);
+}
+
+export function decryptApiKey(apiKey: string): string {
+  if (novuEncrypted(apiKey)) {
+    return decryptSecret(apiKey);
+  }
+
+  return apiKey;
+}
+
+function novuEncrypted(text: string): text is EncryptedSecret {
   return text.startsWith(NOVU_SUB_MASK);
 }
 
-function needEncryption(key: string): boolean {
+function needCredentialEncryption(key: string): boolean {
```

Comment:
Whilst we're changing the name here, I think we should also follow the pattern above and go with the following:
```suggestion
function isCredentialEncryptionRequired(key: string): boolean {
```